homeassistant/components/google_travel_time/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Google Maps Travel Time integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_NAME
+from homeassistant.util import slugify
+
+from .const import (
+    CONF_DESTINATION,
+    CONF_OPTIONS,
+    CONF_ORIGIN,
+    DEFAULT_NAME,
+    DOMAIN,
+    GOOGLE_OPTIONS_SCHEMA,
+    GOOGLE_SCHEMA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Google Maps Travel Time."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize config flow."""
+        self._data = None
+        self._options = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is not None:
+            await self.async_set_unique_id(
+                slugify(
+                    f"{DOMAIN}_{user_input[CONF_ORIGIN]}_{user_input[CONF_DESTINATION]}"
+                )
+            )
+            self._abort_if_unique_id_configured()
```

Comment:
This is a private method right? Isn't it better to use the result of async_set_unique_id to validate if the entry already exists? As in https://github.com/home-assistant/core/pull/42716/files#diff-c40d21d5aecdde1c0388c22771e8b9e3739458659fee3374564a01a8cef106f7R41

Reply:
Why create logic that already exists in a function? This is the same strategy I have used in other integrations